doc/adr/0004-molgenis-core-apps.md

```diff
@@ -0,0 +1,55 @@
+# 4. Molgenis Core Apps
+
+Date: 2020-09-23
+
+## Status
+
+Proposal
+
+## Context
+
+See [frontend architecture sessions](https://docs.google.com/document/d/1VW3ah5VAvAz2KnqNZlNmVqCzFhBMlIcjPPUlsHMFRIY/) for additional background
+information on micro-frontend/SPA architecture.
+
+Molgenis is flexible software that allows users to create their own apps.
+This is quite powerful whenever a new, custom application needs to be made
+on top of the Molgenis backend. However, the same modular paradigm is also
+being applied to parts of the frontend, that are always being shipped with
+Molgenis.
+
+At this time, we have approximately 12 legacy Molgenis apps that are still
+rooted in Molgenis freemarker templates and Java controllers. There are
+about 11 new-style Unpkg frontend applications. The Unpkg applications
+all come with their own package configuration, release cycle, build tools
+and testsuites; all glued together in the Nginx proxy, so they are
+approachable from the same domain.
+
+## Decision
+
+The keyword here is convergence; the act of converging and especially moving
+toward union or uniformity. This means that Molgenis frontend core is going
+to be one Vue application, in which all components from the 11 existing Unpkg
+applications are merged together in a single frontend package. Instead of
+glueing them together with Nginx, we will do this using Vue-router and
+Vuex modules. Instead of having 11 app versions, we will use one frontend
+version and use git tags to mirror each Molgenis release.
```

Comment:
if we have one big spa, then we need to build everything everytime, with the current E2E tests this would lead to build times of over 30 minutes for these tests alone

Reply:
While on this migration path, I think we should go through the E2E testsuite and rewrite what's still useful to the new Playwright setup. E2E tests on a SPA with a tool like Playwright are going to be much faster to execute. However, it could still be that the backend forms a bottleneck. We could either optimize a backend for our E2E tests(e.g. https://playwright.test.molgenis.org/) or defer E2E tests to a once-a-day test, instead of checking on each merge to master.